web/views/reader_view.ex

```diff
@@ -3,6 +3,14 @@ defmodule BlogPhoenix.ReaderView do
   alias BlogPhoenix.Post
   alias BlogPhoenix.ReaderView
 
+  def id(%Post{id: id}) when is_nil(id) do
```

Comment:
This smells bad in the future. It's cruft because of inadequate testing approaches. The tests needs to focus better on what they do so there is no extra code in the main program.